### PR TITLE
Create seperate matrix for selenium test case

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
A lot of time travis CI breaks because of selenium test case, and at that point it becomes trivial to know the exact reason. Hence, in this PR creating a separate matrix for testing selenium test cases.

### What type of PR is it?
Improvement

### Is there a relevant Jira issue?
N/A

### How should this be tested?
Check travis UI. It should have a new matrix added.

### CODE REVIEW

**Comment by prabhjyotsingh** [comment]
Build #3100.4 is failing (https://travis-ci.org/apache/incubator-zeppelin/jobs/108074063) for  

    Results :
    
    Tests in error: 
      CassandraInterpreterTest.org.apache.zeppelin.cassandra.CassandraInterpreterTest » ExceptionInInitializer
      CassandraInterpreterTest.org.apache.zeppelin.cassandra.CassandraInterpreterTest » NoClassDefFound

Which is unrelated.

**Comment by felixcheung** [comment]
@doanduyhai any idea with test failure?

```
DEBUG [17:21:25,783][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE INDEX IF NOT EXISTS idx1 ON live_data.complex_table(indexed1); 
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 36.451 sec <<< FAILURE! - in org.apache.zeppelin.cassandra.CassandraInterpreterTest
org.apache.zeppelin.cassandra.CassandraInterpreterTest  Time elapsed: 36.451 sec  <<< ERROR!
java.lang.ExceptionInInitializerError: null
	at com.datastax.driver.core.exceptions.NoHostAvailableException.copy(NoHostAvailableException.java:84)
	at com.datastax.driver.core.exceptions.NoHostAvailableException.copy(NoHostAvailableException.java:37)
	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
	at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:244)
	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:55)
	at info.archinnov.achilles.script.ScriptExecutor.executeScript(ScriptExecutor.java:61)
	at info.archinnov.achilles.embedded.AchillesInitializer.executeStartupScripts(AchillesInitializer.java:204)
	at info.archinnov.achilles.embedded.AchillesInitializer.initialize(AchillesInitializer.java:98)
	at info.archinnov.achilles.embedded.AchillesInitializer.initializeFromParameters(AchillesInitializer.java:61)
	at info.archinnov.achilles.embedded.CassandraEmbeddedServer.<init>(CassandraEmbeddedServer.java:76)
	at info.archinnov.achilles.embedded.CassandraEmbeddedServerBuilder.buildNativeSessionOnly(CassandraEmbeddedServerBuilder.java:436)
	at org.apache.zeppelin.cassandra.CassandraInterpreterTest.<clinit>(CassandraInterpreterTest.java:59)
Caused by: com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9259 (com.datastax.driver.core.exceptions.OperationTimedOutException: [localhost/127.0.0.1] Timed out waiting for server response))
```

File: zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java [hunk 1]
```diff
@@ -195,53 +195,66 @@ public void testAngularDisplay() throws InterruptedException{
...
-    sparkEditBtn = pollingWait(By.xpath("//div[h3[text()[contains(.,'spark')]]]//button[contains(.,'edit')]"),
-        MAX_BROWSER_TIMEOUT_SEC);
-    sparkEditBtn.click();
-    WebElement testDepRemoveBtn = driver.findElement(By.xpath("//tr[descendant::text()[contains(.,'" +
-        artifact + "')]]/td[3]/div"));
-    sleep(5000, true);
-    testDepRemoveBtn.click();
-    driver.findElement(By.xpath("//button[contains(.,'Save')]")).submit();
-    driver.switchTo().alert().accept();
+    if (!endToEndTestEnabled()) {
```
Comment (felixcheung) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: it looks like this is from PR #654 - could we separate that out for review please?

Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/ZeppelinIT.java]: Yes, its from https://github.com/apache/incubator-zeppelin/pull/673, https://github.com/apache/incubator-zeppelin/pull/673/files#diff-fc9c6a01ec538e646b3e99c8fa19a3deR204

Sure, I've created a new PR https://github.com/apache/incubator-zeppelin/pull/709 as a hot fix.
Will revert this whole file once that is merged. As we will need `if (!endToEndTestEnabled()) {` or else driver `@Before`  will not initialize driver, resulting in this test case to fail.

**Comment by felixcheung** [comment]
Could you check out #708 ? We should avoid rerunning all tests for TEST_SELENIUM

**Comment by prabhjyotsingh** [comment]
Closing this, as this is been taken care by https://github.com/apache/incubator-zeppelin/pull/708

**Comment by felixcheung** [comment]
actually, I think they are addressing different parts of it. It might still make sense to have a separate matrix for just UI tests - to clarify, I was saying you might want to restrict the project list and test scope to avoid running duplicates. Reopen?

**Comment by felixcheung** [comment]
So as you can see, it's re-running all interpreter tests when TEST_SELENIUM="true"
https://s3.amazonaws.com/archive.travis-ci.org/jobs/108461599/log.txt

could you add the `-pl` and `-Dtest` as in https://github.com/apache/incubator-zeppelin/commit/65b47e50549e8c64b5aff9a6ceec16668d1ca52e?

**Comment by doanduyhai** [comment]
@felixcheung 

The error about the CassandraInterpreterTest is because for unit test, I start an embedded Cassandra server and sends CQL queries to test the interpreter.

The exception stack just says that the interpreter cannot connect to the server. It means that the embedded server failed to start for some reason. Can you give me the full stack trace or give me a link to the logs ?

**Comment by felixcheung** [comment]
@doanduyhai job log is here https://travis-ci.org/apache/incubator-zeppelin/jobs/108074063

**Comment by doanduyhai** [comment]
Thanks @felixcheung 

So I had a look into the log file, below is some extract:

```
DEBUG [17:21:08,728][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE KEYSPACE IF NOT EXISTS zeppelin WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':1} AND DURABLE_WRITES=false; 
DEBUG [17:21:08,751][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TABLE IF NOT EXISTS zeppelin.artists( name text PRIMARY KEY, born text, died text, country text, gender text, type text, styles list<text> ); 
DEBUG [17:21:10,068][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TABLE IF NOT EXISTS zeppelin.ts( key text PRIMARY KEY, val text ); 
DEBUG [17:21:11,292][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TABLE IF NOT EXISTS zeppelin.prepared( key text PRIMARY KEY, val text ); 
DEBUG [17:21:12,929][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TYPE IF NOT EXISTS zeppelin.address( street_number int, street_name text, zip_code int, country text, extra_info list<text>, phone_numbers map<text,bigint> ); 
DEBUG [17:21:16,426][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TABLE IF NOT EXISTS zeppelin.users( login text, firstname text, lastname text, age int, deceased boolean, last_update timestamp, addresses frozen<address>, location frozen<tuple<text,bigint,text>>, PRIMARY KEY(login) ); 
DEBUG [17:21:17,848][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE KEYSPACE IF NOT EXISTS samples WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':1} AND DURABLE_WRITES=false; 
DEBUG [17:21:18,994][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE KEYSPACE IF NOT EXISTS live_data WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':1} AND DURABLE_WRITES=false; 
DEBUG [17:21:20,127][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TYPE IF NOT EXISTS live_data.address ( number int, street text, zip int, city text, country text ); 
DEBUG [17:21:21,903][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE TABLE IF NOT EXISTS live_data.complex_table( pk1 uuid, pk2 int, my_static1 text static, my_static2 text static, clustering1 timestamp, clustering2 int, clustering3 text, indexed1 text, indexed2 int, simple double, my_list list<text>, my_udt_list frozen<list<address>>, my_udt frozen<address>, my_map map<int,text>, key_indexed_map map<int,text>, entries_indexed_map map<int,text>, PRIMARY KEY((pk1,pk2),clustering1, clustering2, clustering3) ) WITH CLUSTERING ORDER BY (clustering1 DESC, clustering2 ASC, clustering3 DESC); 
DEBUG [17:21:23,138][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE INDEX IF NOT EXISTS pk2idx ON live_data.complex_table(pk2); 
DEBUG [17:21:24,490][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE INDEX IF NOT EXISTS clustering2idx ON live_data.complex_table(clustering2); 
DEBUG [17:21:25,783][] ACHILLES_DDL_SCRIPT@:executeScript 		SCRIPT : CREATE INDEX IF NOT EXISTS idx1 ON live_data.complex_table(indexed1); 
```

If you look at the timestamp, it took more than 7 secondes to create a few tables in the embedded Cassandra. It means that the embedded Cassandra server was busy because not enough system resources. I guess the Travis CI server was quite busy when we launched the test.

There is really nothing to do other than retry

**Comment by felixcheung** [comment]
@doanduyhai I guess we could check on this - if it has become a frequent problem we could get travis to retry tests. I have not seen this much, on the other hand, selenium test failures seem to be more frequent.

**Comment by prabhjyotsingh** [comment]
@felixcheung made relevant changes, merged with https://github.com/apache/incubator-zeppelin/commit/65b47e50549e8c64b5aff9a6ceec16668d1ca52e, and https://github.com/apache/incubator-zeppelin/pull/709 (master), waiting for CI.

**Comment by felixcheung** [comment]
looks good

**Comment by felixcheung** [comment]
merging today if no more comments

**Pull request closed without merging**